netutils/ftpc/ftpc_putfile.c

```diff
@@ -65,11 +67,60 @@
  *
  ****************************************************************************/
 
+#ifdef CONFIG_FTPC_OVER_SENDFILE
 static int ftpc_sendbinary(FAR struct ftpc_session_s *session,
-                           FAR FILE *linstream, FILE *routstream)
+                           int linfd)
+{
+  struct stat stat_buf;
+  off_t offset;
+  ssize_t result;
+  ssize_t len;
+
+  if (fstat(linfd, &stat_buf) == -1)
+    {
+      ftpc_xfrabort(session, NULL);
+      return ERROR;
+    }
+
+  /* Loop until the entire file is sent */
+
+  offset = session->offset;
+  len = stat_buf.st_size - offset;
+
+  while (len > 0)
+    {
+      result = sendfile(session->data.sd, linfd, &offset, len);
+
+      if (result == -1 && errno == EAGAIN)
+        {
+          continue;
+        }
+      else if (result == -1)
+        {
+          ftpc_xfrabort(session, NULL);
+          return ERROR;
+        }
+
+      len -= result;
+
+      /* Increment the size of the file sent */
+
+      session->size += result;
+    }
+
+  /* Return success */
+
+  return OK;
+}
+
+#else
+
+static int ftpc_sendbinary(FAR struct ftpc_session_s *session,
+                           FAR FILE *linstream)
```

Comment:
why not replace all FILE to int? instead spread #ifdef/#else/#endif around the code base?

Reply:
Do you mean to replace "fopen/fclose/fseek/fflush" in the existing code by "open/close" system calls everywhere?

Reply:
> Do you mean to replace "fopen/fread/fwrite/fclose/fseek/fflush" in the existing code by "open/read/write/close/lseek/none" system calls everywhere?

Look like there are many place call stdio API. Another simple approach is call fileno in ftpc_sendbinary

Reply:
> > Do you mean to replace "fopen/fread/fwrite/fclose/fseek/fflush" in the existing code by "open/read/write/close/lseek/none" system calls everywhere?
> 
> Look like there are many place call stdio API. Another simple approach is call fileno in ftpc_sendbinary

Thank you. I have reworked it using fileno().